add foundry test

0xenj opened the pull request:
- add 3 foundry function test

Commit:
add foundry test

Commit:
add hardhat test

marc-aurele-besner commented on .vscode/settings.json:
```diff
@@ -5,7 +5,7 @@
   "eslint.workingDirectories": [{ "mode": "auto" }],
-  "window.zoomLevel": 2,
-  "editor.tabSize": 2,
+  "window.zoomLevel": 0.5,
```
peut-tu retirer ce fichier de ta PR, aucun problème que tu change les settings localement, mais je ne veut pas les changer dans la main branch

marc-aurele-besner commented on contracts/test/Test-Enzo.t.sol:
```diff
@@ -0,0 +1,81 @@
+  function test_SimpleNft_any_contractUri() public {
+    assertEq(nftContract.contractURI(), '');
+    vm.prank(address(2));
+    nftContract.setContractURI('test contractURI');
```
address(2) ne devrais pas être capable de modifier setContractURI donc il manquerais
```
vm.expectRevert(<message d'erreur>)
```
avant d'apeller setContractURI

ou utiliser verify_revertCall() avec un message d'erreur que nous avons enregistré

marc-aurele-besner commented on test/Test-Enzo.test.js:
```diff
@@ -0,0 +1,199 @@
+
+    console.log('transferOwnership', transferOwnership);
+    const owner_final = await contract.owner();
+    console.log('owner is ', owner_final);
```
rajoute un assertion comme sur ligne 126 pour valider le resultat obtenue

marc-aurele-besner commented on test/Test-Enzo.test.js:
```diff
@@ -0,0 +1,199 @@
+    await contract.setContractURI('Test New ContractURI');
+
+    const contractURI_final = await contract.contractURI();
+    console.log('contractURI_final', contractURI_final);
```
rajoute un assertion comme sur ligne 126 pour valider le resultat obtenue

marc-aurele-besner commented on test/Test-Enzo.test.js:
```diff
@@ -0,0 +1,199 @@
+    await contract.connect(user1).setContractURI('Test New ContractURI');
+
+    const contractURI_final = await contract.contractURI();
+    console.log('contractURI_final', contractURI_final);
```
rajoute un assertion comme sur ligne 126 pour valider le resultat obtenue

marc-aurele-besner commented on test/Test-Enzo.test.js:
```diff
@@ -0,0 +1,199 @@
+    await contract.connect(user2).setBaseURI('TEST');
+
+    const baseURI_final = await contract.baseURI();
+    console.log('baseURI', baseURI_final);
```
rajoute un assertion comme sur ligne 126 pour valider le resultat obtenue

Commit:
clean function : add except remove console.log

marc-aurele-besner commented on test/Test-Enzo.test.js:
```diff
@@ -0,0 +1,196 @@
+  });
+
+  it('Does contract owner can update base uri? (should be)', async function () {
+    expect(await contract.baseURI().to.equal(''));
```
Il y a un erreur de syntax, la bonne syntax est
```
expect(<valeur1>).to.equal(<valeur2>);
```
donc
```
expect(await contract.baseURI()).to.equal('');
```

Corrige toutes les autres lignes avec expect qui ont la même erreur s.v.p.

Commit:
modify and add function

- rectify expect syntax in hardhat test file
- add function test in hardhat test file
- modify foundry test function

0xenj committed:
Delete settings.json

Commit:
add test function hardhat

- delete transferownership test function because we disable the function
- add 2 test function withdraw

Commit:
Merge branch 'Enzo-Test' of https://github.com/marc-aurele-besner/simple-nfterc721 into Enzo-Test
